test/cctest/test_url.cc

```diff
@@ -67,3 +67,28 @@ TEST_F(URLTest, Base3) {
   EXPECT_EQ(simple.host(), "example.org");
   EXPECT_EQ(simple.path(), "/baz");
 }
+
+TEST_F(URLTest, ToFilePath) {
+#define T(url, path) EXPECT_EQ(path, URL(url).ToFilePath())
+  T("http://example.org/foo/bar", "");
+
+#ifdef _WIN32
+  T("file:///C:/Program%20Files/", "C:\\Program Files\\");
+  T("file:///C:/a/b/c?query#fragment", "C:\\a\\b\\c");
+  T("file://host/path/a/b/c?query#fragment", "\\\\host\\path\\a\\b\\c");
+  T("file://xn--weird-prdj8vva.com/host/a", "\\\\wͪ͊eiͬ͋rd.com\\host\\a");
+  T("file:///C:/a%2Fb", "");
```

Comment:
It is a valid URL, since a URL-path-segment string is "zero or more URL units, excluding U+002F (/) and U+003F (?)", where URL units are "URL code points and percent-encoded bytes". percent-encoded bytes that when decoded turn to `/` and `?` don't count.

On the other hand it does lead to confusing behavior, which is why url's getPathFromUrl and the `URL::ToFilePath` I implemented explicitly disallow it, which is why it turns into a `""` (empty = invalid).